prov/sockets/src/sock_conn.c

```diff
@@ -227,7 +227,7 @@ int fd_set_nonblock(int fd)
 	return ret;
 }
 
-#if !defined __APPLE__ && !defined _WIN32
+#if !defined __APPLE__ && !defined __FreeBSD__ && !defined _WIN32
```

Comment:
I'm not an expert in FreeBSD. but I think `IPPROTO_TCP` is synonymous with `SOL_TCP`
Can we do something like this:
```
#define SOL_TCP IPPROTO_TCP
```
in the `include/freebsd/osd.h` file?

Thanks!

Reply:
Good point! I've tested with `#define SOL_TCP IPPROTO_TCP` and I confirm it works!
Thanks, I'll update the patch.